collector/dhcp_collector.go

```diff
@@ -40,44 +57,139 @@ func newDHCPCollector(dhcpClient client.DHCPClient, logger log.Logger) *dhcpColl
 		[]string{"id", "name"},
 		nil,
 	)
+	dhcpAckPacket := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "dhcp", "ack_packet"),
+		"Number of DHCP ACK packets",
+		[]string{"id", "name"},
+		nil,
+	)
+	dhcpDeclinePacket := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "dhcp", "decline_packet"),
+		"Number of DHCP DECLINE packets",
+		[]string{"id", "name"},
+		nil,
+	)
+	dhcpDiscoverPacket := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "dhcp", "discover_packet"),
+		"Number of DHCP DISCOVER packets",
+		[]string{"id", "name"},
+		nil,
+	)
+	dhcpErrorTotal := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "dhcp", "error_total"),
+		"Number of DHCP errors",
+		[]string{"id", "name"},
+		nil,
+	)
+	dhcpInformPacket := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "dhcp", "inform_packet"),
+		"Number of DHCP INFORM packets",
+		[]string{"id", "name"},
+		nil,
+	)
+	dhcpNackPacket := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "dhcp", "nack_packet"),
+		"Number of DHCP NACK packets",
+		[]string{"id", "name"},
+		nil,
+	)
+	dhcpOfferPacket := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "dhcp", "offer_packet"),
+		"Number of DHCP OFFER packets",
+		[]string{"id", "name"},
+		nil,
+	)
+	dhcpReleasePacket := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "dhcp", "release_packet"),
+		"Number of DHCP RELEASE packets",
+		[]string{"id", "name"},
+		nil,
+	)
+	dhcpRequestPacket := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "dhcp", "request_packet"),
+		"Number of DHCP REQUEST packets",
+		[]string{"id", "name"},
+		nil,
+	)
+	dhcpIPPoolSize := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "dhcp", "ip_pool_size_total"),
+		"Total size of dhcp ip pool",
+		[]string{"id", "dhcp_id"},
+		nil,
+	)
+	dhcpIPPoolAllocated := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "dhcp", "ip_pool_allocated_total"),
+		"Total number of allocated ip of dhcp ip pool",
+		[]string{"id", "dhcp_id"},
```

Comment:
should we also include pool id and ip pool range (start, end) if provided?

Reply:
oh nvm just saw this id is actually ip_pool_id

Reply:
It's not provided in statistics